perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -43,8 +44,9 @@
 
 	private final GroupsManagerImplApi groupsManagerImpl;
 	private PerunBl perunBl;
-
-	private Map<Integer, GroupSynchronizerThread> groupSynchronizerThreads;
+	private Integer maxConcurentGroupsToSynchronize;
+	private ConcurrentLinkedDeque<Group> queueOfGroupsToBeSynchronized;
+	private Map<GroupSynchronizerThread, Integer> groupSynchronizerThreads;
```

Comment:
Can you make "queueOfGroupsToBeSynchronized" and "groupSynchronizerThreads" final?